toggle/WebSocksClient.py

```diff
@@ -145,8 +146,10 @@ def run(self):
             self.config.get("Server", "host"), i))
         logging.debug("Websocket connection attempt " + str(i))
         self.connect()
-        ioloop.IOLoop.instance().start()
-        time.sleep(1)
+        self.ioloop = ioloop.IOLoop.instance()
+        self.ioloop.start()
+        if self.state != WebSocksClient.CLOSED:
+          time.sleep(1)
```

Comment:
Much better object oriented design!

Reply:
I think actually I want to revisit this. ioloop.start() is blocking, so if it fails on first attempt, it hangs.